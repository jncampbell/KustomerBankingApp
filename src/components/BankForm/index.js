import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import classNames from 'classnames';
import './styles/bankForm.scss';

class BankForm extends React.Component {
  constructor(props) {
    super(props);
    this.handleTransaction = this.handleTransaction.bind(this);
  }

  handleTransaction() {
    if (this.refs.amount.value) {
      const amount = parseFloat(parseFloat(this.refs.amount.value).toFixed(2));
      const timestamp = moment().format('MMM D, YYYY');
      this.props.processHandler(amount, timestamp);
      this.refs.amount.value = '';
    } else if (this.refs.amount.value < 0) {
      // Handle negative-value error
    } else {
      // Handle no-value error
    }
  }

  render() {
    const buttonClass = classNames(
      `${this.props.type === 'deposit' ?
      'bank-form__deposit-button' :
      'bank-form__withdraw-button'}`
    );
    
    return (
      <div className='bank-form'>
        <input placeholder='Enter amount' type='number' ref='amount' />
        <button className={buttonClass} onClick={this.handleTransaction}>Submit</button>
      </div>
    );
  }
}

BankForm.propTypes = {
  type: PropTypes.string,
  processHandler: PropTypes.func
}

export default BankForm;
