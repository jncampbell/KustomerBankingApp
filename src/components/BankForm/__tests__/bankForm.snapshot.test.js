import React from 'react';
import renderer from 'react-test-renderer';
import BankForm from '../index';

const props = {
  type: 'deposit',
  processHandler: jest.fn()
}

describe('BankForm', () => {
  it('has consistent render output', () => {
    const component = renderer.create(
      <BankForm />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
