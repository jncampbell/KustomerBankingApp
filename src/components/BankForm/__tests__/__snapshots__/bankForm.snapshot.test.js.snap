// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BankForm has consistent render output 1`] = `
<div
  className="bank-form"
>
  <input
    placeholder="Enter amount"
    type="number"
  />
  <button
    className="bank-form__withdraw-button"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;
